packages/esm-patient-registration-app/src/config-schema.ts

```diff
@@ -300,54 +297,65 @@ export const esmPatientRegistrationSchema = {
       ],
       _description:
         'The options for sex selection during patient registration. This is Administrative Gender as it is called by FHIR (Possible options are limited to those defined in FHIR Administrative Gender, see https://hl7.org/fhir/R4/valueset-administrative-gender.html).',
+      _elements: {
+        value: {
+          _type: Type.String,
+          _description:
+            'Value that will be sent to the server. Limited to FHIR-supported values for Administrative Gender',
+          _validators: [validators.oneOf(['male', 'female', 'other', 'unknown'])],
+        },
+        label: {
+          _type: Type.String,
+          _default: '',
+          _description:
+            'The label displayed for the sex option, if it should be different from the value (the value will be translated; the English "translation" is upper-case).',
+        },
+      },
     },
     address: {
       useAddressHierarchy: {
         enabled: {
           _type: Type.Boolean,
-          _description: 'Whether to use the Address hierarchy in the registration form or not',
           _default: true,
+          _description: 'Whether to use the Address hierarchy in the registration form or not',
         },
         useQuickSearch: {
           _type: Type.Boolean,
+          _default: true,
           _description:
             'Whether to use the quick searching through the address saved in the database pre-fill the form.',
-          _default: true,
         },
         searchAddressByLevel: {
           _type: Type.Boolean,
+          _default: false,
           _description:
             "Whether to fill the addresses by levels, i.e. County => subCounty, the current field is dependent on it's previous field.",
-          _default: false,
-        },
-        useAddressHierarchyLabel: {
-          _type: Type.Object,
-          _description: 'Whether to use custom labels for address hierarchy',
-          _default: {},
         },
       },
     },
     dateOfBirth: {
       allowEstimatedDateOfBirth: {
         _type: Type.Boolean,
-        _description: 'Whether to allow estimated date of birth for a patient during registration',
         _default: true,
+        _description: 'Whether to allow estimated date of birth for a patient during registration',
       },
       useEstimatedDateOfBirth: {
         enabled: {
           _type: Type.Boolean,
-          _description: 'Whether to use a fixed day and month for estimated date of birth',
           _default: false,
+          _description: 'Whether to use a fixed day and month for estimated date of birth',
         },
         dayOfMonth: {
           _type: Type.Number,
-          _description: 'The custom day of the month use on the estimated date of birth',
           _default: 0,
+          _description: 'The custom day of the month use on the estimated date of birth',
+          _validators: [validators.inRange(0, 30)],
```

Comment:
```suggestion
          _description: 'The custom day of the month use on the estimated date of birth  (0 = last day of previous month, 1-31 = specific day)',
          _validators: [validators.inRange(0, 31)],
```

The range (0, 31) matches JavaScript's Date constructor behavior used in `calcBirthdate()`:

  - 0 is valid: new Date(year, month, 0) means "last day of previous month" - allows setting estimated births to end of previous month
  - 1-31 covers all valid days: JavaScript handles month overflow gracefully (Feb 31st → March 2nd)
  - Original (0, 30) was too restrictive: Would incorrectly reject valid day 31

This ensures the config accepts all meaningful date values while preventing invalid inputs that could break date calculations.

Reply:
Additionally, the test in `patient-registration.test.tsx` is using the wrong `Date` method to compute [dayOfMonth](https://github.com/openmrs/openmrs-esm-patient-management/blob/main/packages/esm-patient-registration-app/src/patient-registration/patient-registration.test.tsx#L136). Since dayOfMonth represents "day of the month" in the birth date calculation, the test should use `getDate()` to get a realistic day-of-month value (1-31) rather than a day-of-week value (0-6).

Without this fix, the test would set dayOfMonth to 0-6 instead of 1-31, which works but doesn't reflect real-world usage where users would configure actual calendar days.